### DISCUSSION TITLE
Allow Snaps to add a new tab in the UI

### DISCUSSION THREAD

**Discussion by PaulRBerg** [MetaMask/snaps#1857] [Ideas]
The [FAQ](https://support.metamask.io/hc/en-us/articles/18245938714395-MetaMask-Snaps-FAQ) says this:

> Can Snaps be used to modify the user interface or user experience of MetaMask?
> Snaps can provide custom user interfaces in dialogs and in the transaction confirmation flow.

This is fairly limiting.

We were thinking of implementing a MetaMask Snap for [Sablier](https://sablier.com) that would have two features:

1. Display the list of token balances that are available for withdrawal from Sablier
2. Allow the user to withdraw from all streams our [`withdrawMultiple`](https://github.com/sablier-labs/v2-core/blob/6189bbc099d1d5fc50b62b93fd9a5fcfbadd7144/src/abstracts/SablierV2Lockup.sol#L330-L338) function.

However, we cannot do this because there's no good place where the list of token balances can be displayed. I gather that we could hack something with an [alert dialog](https://docs.metamask.io/snaps/reference/rpc-api/#alert-dialog), but that seems misguided, and there isn't enough screen real estate anyway. For the Sablier Snap to be functional, we need a dedicated tab.

Would you be open to enabling this functionality? Cc @Montoya, @danfinlay.

Related:

- https://github.com/MetaMask/snaps/discussions/1732
- https://github.com/MetaMask/snaps/discussions/1285

Keywords for search: custom, custom user interface, custom UI, dialog, snap_dialog

**Comment by Montoya** [comment]
Yes we are working on something to enable what you want to do. Keep an eye on https://github.com/MetaMask/SIPs for updates on this.

**Reply by PaulRBerg** [reply]
Awesome, glad to hear
